java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/ClientCredentialsGrantTokenRequestBuilder.java

```diff
@@ -152,7 +154,20 @@ protected String buildJwtAssertion(String clientId, String audience) {
     }
 
     protected void setJWTCustomHeaders(JsonWebSignature.Header header) {
-        header.setX509Thumbprint(encodeKeyId(secretStore.getConfigPropertyOrError(ConfigProperty.PRIVATE_KEY_ID)));
+        String configuredValue = secretStore.getConfigPropertyOrError(ConfigProperty.PRIVATE_KEY_ID);
+
+        // these private key ids are fingerprints generated by openssl tooling. we've seen on
+        // some systems that fingerprints end up with extra garbage.
+        configuredValue = trimPrefixIfPresent(configuredValue, "sha1 Fingerprint=");
+        header.setX509Thumbprint(encodeKeyId(configuredValue));
+    }
+
+    @VisibleForTesting
+    String trimPrefixIfPresent(@Nullable String s, @NonNull String prefix) {
```

Comment:
Use `StringUtils.removeStart` or (removeStartIgnoreCase) - already in use in the project and all methods null safe